src/client.rs

```diff
@@ -500,77 +483,73 @@ pub(crate) unsafe extern "C" fn native_error_cb<C: ClientContext>(
     context.error(error, reason.trim());
 }
 
-unsafe fn handle_refresh_error_msg(client: *mut RDKafka, error_msg: &str) {
-    error!("{}", error_msg);
-    rdkafka_sys::rd_kafka_oauthbearer_set_token_failure(
-        client,
-        error_msg.as_ptr() as *const c_char,
-    );
+/// A generated OAuth token and its associated metadata.
+///
+/// When using the `OAUTHBEARER` SASL authentication method, this type is
+/// returned from [`ClientContext::refresh_oauth_token`]. The token and
+/// principal name must not contain embedded null characters.
+///
+/// Specifying SASL extensions is not currently supported.
+pub struct OAuthToken {
```

Comment:
If/when SASL extensions support is added, is the plan to add another `pub` field to this struct? If so, then adding support for SASL extensions will break the public-facing API for this struct, which I'd rather not happen.

If I'm wrong, please let me know what your plan is. Otherwise, I think maybe we keep the fields `pub(crate)` at most, and use `OAuthToken::new(token: String, principal_name: String, lifetime_ms: i64)` as the public api.

Reply:
We're still pre-1.0, so we're free to make breaking changes at any time. I'm going to leave as is since it's quite a bit more convenient to have the named fields, and we can revisit if we ever get close to shipping a 1.0.